easybuild/easyblocks/i/icc.py

```diff
@@ -79,87 +95,129 @@ def install_step(self):
     def sanity_check_step(self):
         """Custom sanity check paths for icc."""
 
-        binprefix = "bin/intel64"
-        libprefix = "lib/intel64/lib"
-        if LooseVersion(self.version) >= LooseVersion("2011"):
-            if LooseVersion(self.version) <= LooseVersion("2011.3.174"):
-                binprefix = "bin"
-            elif LooseVersion(self.version) >= LooseVersion("2013_sp1"):
-                binprefix = "bin"
-                libprefix = "lib/intel64/lib"
+        binprefix = 'bin/intel64'
+        libprefix = 'lib/intel64'
+        if LooseVersion(self.version) >= LooseVersion('2011'):
+            if LooseVersion(self.version) <= LooseVersion('2011.3.174'):
+                binprefix = 'bin'
+            elif LooseVersion(self.version) >= LooseVersion('2013_sp1'):
+                binprefix = 'bin'
             else:
-                libprefix = "compiler/lib/intel64/lib"
+                libprefix = 'compiler/lib/intel64'
 
-        binfiles = ["icc", "icpc"]
-        if LooseVersion(self.version) < LooseVersion("2014"):
-            binfiles += ["idb"]
+        binfiles = ['icc', 'icpc']
+        if LooseVersion(self.version) < LooseVersion('2014'):
+            binfiles += ['idb']
+
+        binaries = [os.path.join(binprefix, f) for f in binfiles]
+        libraries = [os.path.join(libprefix, 'lib%s' % l) for l in ['iomp5.a', 'iomp5.so']]
+        sanity_check_files = binaries + libraries
+        if LooseVersion(self.version) > LooseVersion('2015'):
+            sanity_check_files.append('include/omp.h')
 
         custom_paths = {
-            'files': ["%s/%s" % (binprefix, x) for x in binfiles] +
-                     ["%s%s" % (libprefix, x) for x in ["iomp5.a", "iomp5.so"]],
+            'files': sanity_check_files,
             'dirs': [],
         }
 
         super(EB_icc, self).sanity_check_step(custom_paths=custom_paths)
 
     def make_module_req_guess(self):
-        """Customize paths to check and add in environment.
         """
+        Additional paths to consider for prepend-paths statements in module file
+        """
+        prefix = None
+
+        # guesses per environment variables
+        # some of these paths only apply to certain versions, but that doesn't really matter
+        # existence of paths is checked by module generator before 'prepend-paths' statements are included
+        guesses = {
+            'CLASSPATH': ['daal/lib/daal.jar'],
+            # 'include' is deliberately listed last, so it gets precedence over the other subdirs
+            'CPATH': ['daal/include', 'ipp/include', 'mkl/include', 'tbb/include', 'include'],
```

Comment:
`$CPATH` was not set before, and it seems like we shouldn't start setting it either...

I think this is causing the problem in discussed at https://github.com/hpcugent/easybuild-easyblocks/pull/772/files#r47450112, i.e. error like this popping up:

```
lib/types.h(107): error: qualified name is not allowed                                                                                                                         
    typedef std::complex<double> cdouble;                                                                                                                                      
...                                                                                                                    
lib/types.h(107): error: expected a ";"                                                                                                                                        
    typedef std::complex<double> cdouble;  
```

unless the `include/c++/4.9.3` subdirectory of the GCC install dir is prepended to `$CPATH`

this is also mentioned in https://software.intel.com/en-us/forums/intel-c-compiler/topic/338378, to quote:

```
The problem has been resolved by now. Our system admin, who installed the compiler using the standard Intel installer (only changed the install directory from the default, but nothing else), unnecessarily added the the Intel include paths to the environment variables CPLUS_INCLUDE_PATH, so that #include_next included the same file again (like me, he wasn't aware of the #include_next directive).
```